src/VisualStudio/Core/Def/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -135,22 +142,12 @@ public IFileChangeContext CreateContext(params WatchedDirectory[] watchedDirecto
             /// <summary>
             /// A collection of file watcher tokens to remove for <see cref="Kind.UnwatchFiles"/>.
             /// </summary>
-            private readonly IEnumerable<Context.RegularWatchedFile> _tokens;
+            private readonly ImmutableArray<Context.RegularWatchedFile> _tokens;
 
-            private WatcherOperation(Kind kind)
-            {
-                Contract.ThrowIfFalse(kind is Kind.None);
-                _kind = kind;
-
-                // Other watching fields are not used for this kind
-                _directory = null!;
-                _filter = null;
-                _fileChangeFlags = 0;
-                _sink = null!;
-                _cookies = null!;
-                _token = null!;
-                _tokens = null!;
-            }
+            /// <summary>
+            /// A collection of file paths to subscribe to for <see cref="Kind.WatchFiles"/>/>.
+            /// </summary>
+            private readonly ImmutableArray<string> _files;
```

Comment:
Since we're using an ImmutableArray here, would it make sense just to make this be a [OneOrMany](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs) and then get rid of the field that is holding the single watch case? Ditto for _token. At that point we can just get rid of "WatchFile" as a state entirely, _only_ have WatchFiles, and if you want to be clever just special-case the single-file case and call the old method if you really want to.